packages/sdk/src/modules/rbac/index.ts

```diff
@@ -517,35 +529,26 @@ export class RBAC extends Base {
 
   /**
    * Fetch all roles.
-   * @param ownerAddress - The ownerAddress is public address of user or owner who created a roles.
+   * @param options - The ownerAddress is public address of user or owner who created a roles.
    * @returns A promise that resolves when the role is fetched.
    */
 
-  public async fetchRoles(ownerAddress: Address): Promise<FetchRoles[]> {
+  public async fetchRoles(options: FetchRoles): Promise<ResponseFetchGroup[]> {
     try {
-      if (!ownerAddress) throw new Error('Invalid owner address');
+      const { owner } = options;
+      if (!owner) throw new Error('Invalid owner address');
       const api = this._getApi();
       const roles = (await api.query?.['peaqRbac']?.['roleStore'](
-        ownerAddress
+        owner
       )) as unknown as Entity[];
       if (!roles) {
         throw new Error(
-          `Roles not exits with this owner address = ${ownerAddress}`
+          `Roles not exits with this owner address = ${owner}`
         );
       }
-      const responseData: FetchRoles[] = [];
-      roles.forEach((item, index) => {
-        const readAbleData = item.toHuman();
-        const stringFyData = JSON.stringify(readAbleData);
-        const parseData = JSON.parse(stringFyData);
-        const { id, name, enabled } = parseData;
-        let payload = {
-          id,
-          name,
-          enabled,
-        };
-        responseData.push(payload);
-      });
+      const responseData: ResponseFetchGroup[] = roles?.map(
+        (item) => JSON.parse(JSON.stringify(item.toHuman()))
```

Comment:
when is it necessary to use the hashed_key in order to read?

Reply:
Do I need to use a hash key here or is the owner address fine to read all the rbac roles associated with the caller's address? It seems like roleStore only works when the owner address passed, not a key. @rohanranjan250613 @lavish0000